src/test_lib_json/fuzz.h

```diff
@@ -0,0 +1,14 @@
+// Copyright 2007-2010 The JsonCpp Authors
+// Distributed under MIT license, or public domain if desired and
+// recognized in your jurisdiction.
+// See file LICENSE for detail or copy at http://jsoncpp.sourceforge.net/LICENSE
+
+#ifndef FUZZ_H_INCLUDED
+#define FUZZ_H_INCLUDED
+
+#include <stddef.h>
+#include <stdint.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size);
```

Comment:
Do we actually need extern "C"? I thought this was all C++ code.

Reply:
Yes, we do need it so that it can find LLVM.

Reply:
This is still failing to get linked properly on some platforms. I've updated the build bots to be more modern, which might help remove this issue. If we fix the compile problem I think this patch is good to go.

```
jsoncpp_test@exe/src_test_lib_json_main.cpp.o: In function `TestFuzzTestfuzzDoesntCrash::runTestCase()':
../src/test_lib_json/main.cpp:(.text+0x2c08a): undefined reference to `LLVMFuzzerTestOneInput'
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Reply:
Do you know how I can fix this compile problem? I couldn't find the bug for this error.

Reply:
Not sure, but I do see it locally building with meson + ninja:

```
FAILED: jsoncpp_test 
c++  -o jsoncpp_test 'jsoncpp_test@exe/src_test_lib_json_jsontest.cpp.o' 'jsoncpp_test@exe/src_test_lib_json_main.cpp.o' -Wl,-dead_strip_dylibs libjsoncpp.21.dylib -Wl,-headerpad_max_install_names -Wl,-rpath,@loader_path/
Undefined symbols for architecture x86_64:
  "_LLVMFuzzerTestOneInput", referenced from:
      TestFuzzTestfuzzDoesntCrash::runTestCase() in src_test_lib_json_main.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Reply:
Figured it out. We don't build with CMake--that's a nicety for some of our clients. Internally we use the meson+ninja build outlined in the docs, which require updating a different build file, the meson.build. The test compiles successfully with the below change:

```
-> % git diff
diff --git a/meson.build b/meson.build
index ece8aa4..8de947c 100644
--- a/meson.build
+++ b/meson.build
@@ -88,7 +88,8 @@ jsoncpp_test = executable(
   'jsoncpp_test',
   [ 'src/test_lib_json/jsontest.cpp',
     'src/test_lib_json/jsontest.h',
-    'src/test_lib_json/main.cpp'],
+    'src/test_lib_json/main.cpp',
+    'src/test_lib_json/fuzz.cpp'],
   include_directories : jsoncpp_include_directories,
   link_with : jsoncpp_lib,
   install : false,
```

Reply:
Thanks! It passed!

Reply:
It can be merged.